src/spikeinterface/extractors/iblextractors.py

```diff
@@ -0,0 +1,289 @@
+            adc = np.concatenate((electrodes_geometry["adc"], [np.nan]))
+            index_on_probe = np.concatenate((electrodes_geometry["ind"], [np.nan]))
+            good_channel = np.concatenate((electrodes_geometry["shank"], [1.0]))
+
+        self.set_property("shank", shank)
+        self.set_property("shank_row", shank_row)
+        self.set_property("shank_col", shank_col)
+        self.set_property("inter_sample_shift", inter_sample_shift)
+        self.set_property("adc", adc)
+        self.set_property("index_on_probe", index_on_probe)
+        if not all(good_channel):
+            self.set_property("good_channel", good_channel)
+
+        # init recording segment
+        recording_segment = IblRecordingSegment(file_streamer=self._file_streamer, load_sync_channel=load_sync_channel)
+        self.add_recording_segment(recording_segment)
+
+        self._kwargs = {
+            "session": session,
+            "stream_name": stream_name,
+            "load_sync_channel": load_sync_channel,
+            "cache_folder": cache_folder,
+            "remove_cached": remove_cached,
+            "stream": stream,
+        }
+
+
+class IblRecordingSegment(BaseRecordingSegment):
+    def __init__(self, file_streamer, load_sync_channel: bool = False):
+        BaseRecordingSegment.__init__(self, sampling_frequency=file_streamer.fs)
+        self._file_streamer = file_streamer
+        self._load_sync_channel = load_sync_channel
+
+    def get_num_samples(self):
+        return self._file_streamer.ns
+
+    def get_traces(self, start_frame: int, end_frame: int, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+        if channel_indices is None:
+            channel_indices = slice(None)
+        traces = self._file_streamer.read(nsel=slice(start_frame, end_frame), volts=False)
+        if not self._load_sync_channel:
+            traces = traces[:, :-1]
+
+        return traces[:, channel_indices]
+
+
+class IblSortingExtractor(BaseSorting):
+    """Load IBL data as a sorting extractor.
+
+    Parameters
+    ----------
+    one: One = None
+    pid: str = None
+    eid: str = ''
+    pname: str = ''
+
+    Returns
+    -------
+    extractor : IBLSortingExtractor
+        The loaded data.
+    """
+
+    extractor_name = "IBLSorting"
+    name = "ibl"
+    installation_mesg = "IBL extractors require ibllib as a dependency." " To install, run: \n\n pip install ibllib\n\n"
+
+    def __init__(self, one=None, pid=None, eid="", pname="", **kwargs):
+        # check correct parent folder:
+        try:
+            from one.api import ONE
+            from brainbox.io.one import SpikeSortingLoader
+        except ImportError:
+            raise ImportError(self.installation_mesg)
+        self.ssl = SpikeSortingLoader(one=one, pid=pid, eid=eid, pname=pname, **kwargs)
```

Comment:
Okay, so here the best is normally to download first the data and then run multiprocessing. Streaming is by definition not a HPC solution.

But since people still do it, we have to support this. We really do not want to instantiate a default ONE object, this practice has large drawbacks and obfuscate some error sources.

I suggest to feed a dict instead as in : 506a45c54e7080cbb8e99b8993e17ca893af4e08
This makes sure all spawned processes are instantiated the same way.